### PR TITLE
Improve block build time calculation

### PR DESCRIPTION
**Pull request by smiasojed**
Previously, it checked the time offset between two blocks. Now, it calculates the time offset between the first and last block and divides it by the number of blocks.

### CODE REVIEW

File: src/stats.rs [hunk 1]
```diff
@@ -84,16 +84,14 @@ pub async fn print_block_info(
                 call_block_expected = true;
             }
             if call_block_expected {
+                time_stamp_last = Some(block.time_stamp);
+                if time_stamp_first.is_none() {
+                    time_stamp_first = time_stamp_last;
```
Comment (ascjones) [inline src/stats.rs:89]: Looks like `time_stamp_first` will always be the same as `time_stamp_last`?

Comment (smiasojed) [reply src/stats.rs:89]: Only for the first block, then it should be != None and will not be updated

Comment (smiasojed) [reply src/stats.rs:89]: It should be the same only for the first block in the loop

Comment (ascjones) [reply src/stats.rs:89]: :facepalm:

**Review by ascjones — APPROVED** [review]

**Pull request merged**